client/components/domains/map-domain-step/index.jsx

```diff
@@ -20,16 +20,17 @@ var MapDomainStep = React.createClass( {
 
 	propTypes: {
 		products: React.PropTypes.object.isRequired,
+		initialQuery: React.PropTypes.string,
 		analyticsSection: React.PropTypes.string.isRequired
 	},
 
 	getInitialState: function() {
-		return { searchQuery: '' };
+		return { searchQuery: this.props.initialQuery };
 	},
 
 	componentWillMount: function() {
 		if ( this.props.initialState ) {
-			this.setState( this.props.initialState );
+			this.setState( Object.assign( {}, this.props.initialState, this.getInitialState() ) );
```

Comment:
This seems... backwards.

Why do we have to get an `initialState` from the props and then set that to the state itself using `setState`?

I know that you didn't started it but I'd love to get rid of it. If it's easy.